Mathlib/CategoryTheory/GradedObject/Monoidal.lean

```diff
@@ -271,10 +296,351 @@ lemma associator_naturality (f₁ : X₁ ⟶ Y₁) (f₂ : X₂ ⟶ Y₂) (f₃
+  [∀ (X₁ X₂ X₃ : GradedObject I C), HasGoodTensorTensor₂₃ X₁ X₂ X₃]
+  [DecidableEq I] [HasInitial C]
+  [∀ X₁, PreservesColimit (Functor.empty.{0} C) ((curriedTensor C).obj X₁)]
+  [∀ X₂, PreservesColimit (Functor.empty.{0} C) ((curriedTensor C).flip.obj X₂)]
+  [∀ (X₁ X₂ X₃ X₄ : GradedObject I C), HasTensor₄ObjExt X₁ X₂ X₃ X₄]
+
+noncomputable instance monoidalCategory : MonoidalCategory (GradedObject I C) where
+  tensorObj X Y := Monoidal.tensorObj X Y
+  tensorHom f g := Monoidal.tensorHom f g
+  tensorHom_def f g := Monoidal.tensorHom_def f g
+  whiskerLeft X _ _ φ := Monoidal.whiskerLeft X φ
+  whiskerRight {_ _ φ Y} := Monoidal.whiskerRight φ Y
+  tensorUnit := Monoidal.tensorUnit
+  associator X₁ X₂ X₃ := Monoidal.associator X₁ X₂ X₃
+  associator_naturality f₁ f₂ f₃ := Monoidal.associator_naturality f₁ f₂ f₃
+  leftUnitor X := Monoidal.leftUnitor X
+  leftUnitor_naturality := Monoidal.leftUnitor_naturality
+  rightUnitor X := Monoidal.rightUnitor X
+  rightUnitor_naturality := Monoidal.rightUnitor_naturality
+  tensor_comp f₁ f₂ g₁ g₂ := Monoidal.tensor_comp f₁ g₁ f₂ g₂
+  pentagon X₁ X₂ X₃ X₄ := Monoidal.pentagon X₁ X₂ X₃ X₄
+  triangle X₁ X₂ := Monoidal.triangle X₁ X₂
+
+/-variable {A : C} (X₁ X₂ X₃ X₄ Y₁ Y₂ : GradedObject I C)
+
+noncomputable def tensorObjIso :
+    X₁ ⊗ X₂ ≅ mapBifunctorMapObj (curriedTensor C) (fun ⟨i, j⟩ => i + j) X₁ X₂ := Iso.refl _
+
+noncomputable def ιTensorObj (i₁ i₂ i₁₂ : I) (h : i₁ + i₂ = i₁₂) :
+  X₁ i₁ ⊗ X₂ i₂ ⟶ (X₁ ⊗ X₂) i₁₂ :=
+    Monoidal.ιTensorObj X₁ X₂ i₁ i₂ i₁₂ h
+
+variable {X₁ X₂ Y₁ Y₂}
+
+@[reassoc (attr := simp)]
+lemma ι_tensorHom (f : X₁ ⟶ X₂) (g : Y₁ ⟶ Y₂) (i₁ i₂ i₁₂ : I) (h : i₁ + i₂ = i₁₂) :
+    ιTensorObj X₁ Y₁ i₁ i₂ i₁₂ h ≫ tensorHom f g i₁₂ =
+      (f i₁ ⊗ g i₂) ≫ ιTensorObj X₂ Y₂ i₁ i₂ i₁₂ h := by
+  apply Monoidal.ι_tensorHom
+
+variable (X₁ X₂)
+
+@[simp]
+abbrev cofanTensorFun (j : I) : { i : I × I | i.1 + i.2 = j } → C :=
+  fun ⟨⟨i₁, i₂⟩, _⟩ => X₁ i₁ ⊗ X₂ i₂
+
+@[simp]
+noncomputable def cofanTensor (j : I) : Cofan (cofanTensorFun X₁ X₂ j) :=
+  Cofan.mk ((X₁ ⊗ X₂) j) (fun ⟨⟨i₁, i₂⟩, hi⟩ => ιTensorObj X₁ X₂ i₁ i₂ j hi)
+
+noncomputable def isColimitCofanTensor (j : I) : IsColimit (cofanTensor X₁ X₂ j) := by
+  apply isColimitCofanMapObj
+
+variable {X₁ X₂}
+
+noncomputable def descTensor {j : I} (f : ∀ (i₁ i₂ : I) (_ : i₁ + i₂ = j), X₁ i₁ ⊗ X₂ i₂ ⟶ A) :
+    (X₁ ⊗ X₂) j ⟶ A :=
+  Cofan.IsColimit.desc (isColimitCofanTensor X₁ X₂ j) (fun ⟨⟨i₁, i₂⟩, hi⟩ => f i₁ i₂ hi)
+
+@[reassoc (attr := simp)]
+lemma ι_descTensor (j : I) (f : ∀ (i₁ i₂ : I) (_ : i₁ + i₂ = j), X₁ i₁ ⊗ X₂ i₂ ⟶ A)
+    (i₁ i₂ : I) (hi : i₁ + i₂ = j) :
+    ιTensorObj X₁ X₂ i₁ i₂ j hi ≫ descTensor f = f i₁ i₂ hi := by
+  apply Cofan.IsColimit.fac
+
+@[ext]
+lemma tensorObj_ext {j : I} (f g : (X₁ ⊗ X₂) j ⟶ A)
+    (h : ∀ (i₁ i₂ : I) (hi : i₁ + i₂ = j),
+      ιTensorObj X₁ X₂ i₁ i₂ j hi ≫ f = ιTensorObj X₁ X₂ i₁ i₂ j hi ≫ g) : f = g :=
+  Monoidal.tensorObj_ext f g h-/
```

Comment:
Do you intend to keep this in the PR? Should it be un-commented?

Reply:
I have removed this for now. In the future, we may have to develop such an API (which would basically just duplicate the API of the internals of the construction of the monoidal category instance).